src/workerd/server/server.c++

```diff
@@ -1447,9 +1449,183 @@ void Server::InspectorServiceIsolateRegistrar::registerIsolate(
 
 // =======================================================================================
 namespace {
+
+// The TailStreamWriterState holds the current client-side state for a collection
+// of streaming tail workers that a worker is reporting events to.
+struct TailStreamWriterState {
```

Comment:
Non-blocking: Maybe there's some synergies with upstream here and this could live in a seperate file to avoid code duplication, that can be handled in the future though.

Reply:
Yep. Already in plan. I'll add a comment.